mdc_weight_mgmt_xml_import/models/mdc_weight_record.py

```diff
@@ -0,0 +1,134 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import models, api, _
+from datetime import datetime
+from lxml import etree as ET
+from pathlib import Path
+import traceback
+import logging
+import re
+
+class MdcWeightRecord(models.Model):
+    _inherit = 'mdc.weight.record'
+
+    @api.model
+    def cron_import_xml_weight(self):
+        self._read_qc_inspection_xml()
+
+    def _read_qc_inspection_xml(self):
+        logger = logging.getLogger(__name__)
+        raw_in_path = self.env.company.xml_weight_in_path
+        raw_processed_path = self.env.company.xml_weight_processed_path
+
+        if not raw_in_path or not raw_processed_path:
+            logger.error("XML paths not defined in company settings")
+            return
+
+        in_path = Path(raw_in_path)
+        processed_path = Path(raw_processed_path)
+
+        try:
+            in_path.mkdir(parents=True, exist_ok=True)
+            processed_path.mkdir(parents=True, exist_ok=True)
+        except OSError as e:
+            logger.error("Can't create folders: %s", e)
+            return
+
+        try:
+            files = [f for f in in_path.iterdir() if f.is_file()]
+        except Exception as e:
+            logger.error("Error listing files: %s", e)
+            return
+
+        if not files:
+            logger.info(_("No files to process in %s", in_path))
+            return
+
+        for file in files:
+            if not file.name.lower().endswith('.xml'):
+                logger.warning("Skipping non-XML file: %s", file.name)
+                continue
+            try:
+                self._process_xml_file(file, processed_path)
+            except Exception as e:
+                self.env['mdc.weight.record.error'].create({
```

Comment:
If we use sudo when creating any record of this, we can remove creation permission access for every user (nobody should manually create a record for this)
```suggestion
                self.env['mdc.weight.record.error'].sudo().create({
```